internal/provider/resource_nd_multi_cluster_connectivity.go

```diff
@@ -0,0 +1,652 @@
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+	tflog.Debug(ctx, "End configure of resource: nd_multi_cluster_connectivity")
+}
+
+func (r *ClusterResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Debug(ctx, "Start create of resource: nd_multi_cluster_connectivity")
+
+	var stateData *ClusterResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &stateData)...)
+
+	var data *ClusterResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	jsonPayload := getClusterCreateJsonPayload(ctx, &resp.Diagnostics, data, "POST")
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	r.client.DoRestRequest(ctx, &resp.Diagnostics, clusterPath, "POST", jsonPayload)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	setClusterId(ctx, data, data.FabricName.ValueString())
+	getAndSetClusterAttributes(ctx, &resp.Diagnostics, r.client, data)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	tflog.Debug(ctx, fmt.Sprintf("End create of resource nd_multi_cluster_connectivity with id '%s'", data.Id.ValueString()))
+}
+
+func (r *ClusterResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Debug(ctx, "Start read of resource: nd_multi_cluster_connectivity")
+	var data *ClusterResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Read of resource nd_multi_cluster_connectivity with id '%s'", data.Id.ValueString()))
+
+	getAndSetClusterAttributes(ctx, &resp.Diagnostics, r.client, data)
+
+	// Save updated data into Terraform state
+	if data.Id.IsNull() {
+		var emptyData *ClusterResourceModel
+		resp.Diagnostics.Append(resp.State.Set(ctx, &emptyData)...)
+	} else {
+		resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("End read of resource nd_multi_cluster_connectivity with id '%s'", data.Id.ValueString()))
+}
+
+func (r *ClusterResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Debug(ctx, "Start update of resource: nd_multi_cluster_connectivity")
+
+	var stateData *ClusterResourceModel
+	var data *ClusterResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Update of resource nd_multi_cluster_connectivity with id '%s'", data.Id.ValueString()))
+
+	jsonPayload := getClusterCreateJsonPayload(ctx, &resp.Diagnostics, data, "PUT")
```

Comment:
why call the function getClusterCreateJsonPayload when it can also be used for update?